op-dispute-mon/mon/detector_test.go

```diff
@@ -0,0 +1,434 @@
+		creator.loader = loader
+		_, _, status, err := detector.fetchGameMetadata(context.Background(), types.GameMetadata{})
+		require.NoError(t, err)
+		require.Equal(t, types.GameStatusInProgress, status)
+	})
+}
+
+func TestDetector_CheckAgreement_Fails(t *testing.T) {
+	detector, _, _, rollup := setupDetectorTest(t)
+	rollup.err = errors.New("boom")
+	_, err := detector.checkAgreement(context.Background(), common.Address{}, 0, common.Hash{}, types.GameStatusInProgress)
+	require.ErrorIs(t, err, rollup.err)
+}
+
+func TestDetector_CheckAgreement_Succeeds(t *testing.T) {
+	tests := []struct {
+		name        string
+		rootClaim   common.Hash
+		status      types.GameStatus
+		expectBatch func(*detectionBatch)
+		err         error
+	}{
+		{
+			name: "in_progress",
+			expectBatch: func(batch *detectionBatch) {
+				require.Equal(t, 1, batch.inProgress)
+			},
+		},
+		{
+			name:      "agree_defender_wins",
+			rootClaim: mockRootClaim,
+			status:    types.GameStatusDefenderWon,
+			expectBatch: func(batch *detectionBatch) {
+				require.Equal(t, 1, batch.agreeDefenderWins)
+			},
+		},
+		{
+			name:   "disagree_defender_wins",
+			status: types.GameStatusDefenderWon,
+			expectBatch: func(batch *detectionBatch) {
+				require.Equal(t, 1, batch.disagreeDefenderWins)
+			},
+		},
+		{
+			name:      "agree_challenger_wins",
+			rootClaim: mockRootClaim,
+			status:    types.GameStatusChallengerWon,
+			expectBatch: func(batch *detectionBatch) {
+				require.Equal(t, 1, batch.agreeChallengerWins)
+			},
+		},
+		{
+			name:   "disagree_challenger_wins",
+			status: types.GameStatusChallengerWon,
+			expectBatch: func(batch *detectionBatch) {
+				require.Equal(t, 1, batch.disagreeChallengerWins)
+			},
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			detector, _, _, _ := setupDetectorTest(t)
+			batch, err := detector.checkAgreement(context.Background(), common.Address{}, 0, test.rootClaim, test.status)
+			require.NoError(t, err)
+			test.expectBatch(&batch)
+		})
+	}
+}
+
+func TestDetector_CheckRootAgreement(t *testing.T) {
+	t.Parallel()
+
+	t.Run("OutputFetchFails", func(t *testing.T) {
+		detector, _, _, rollup := setupDetectorTest(t)
+		rollup.err = errors.New("boom")
+		agree, err := detector.checkRootAgreement(context.Background(), 0, mockRootClaim)
+		require.ErrorIs(t, err, rollup.err)
+		require.False(t, agree)
+	})
+
+	t.Run("OutputMismatch", func(t *testing.T) {
+		detector, _, _, _ := setupDetectorTest(t)
+		agree, err := detector.checkRootAgreement(context.Background(), 0, common.Hash{})
+		require.NoError(t, err)
+		require.False(t, agree)
+	})
+
+	t.Run("OutputMatches", func(t *testing.T) {
+		detector, _, _, _ := setupDetectorTest(t)
+		agree, err := detector.checkRootAgreement(context.Background(), 0, mockRootClaim)
+		require.NoError(t, err)
+		require.True(t, agree)
+	})
+}
+
+// func TestDetector_ProcessGame(t *testing.T) {
```

Comment:
Why are these left commented out? Should they be deleted or uncommented?

Reply:
Shoot my mistake, accidental commit!